packages/kuma-gui/src/app/application/components/data-source/DataLoader.vue

```diff
@@ -51,22 +20,27 @@
         />
       </slot>
     </template>
-    <template v-else>
+    <template
+      v-else-if="allData.length > 0 && allData.every(item => typeof item !== 'undefined')"
+    >
       <slot
-        name="loadable"
-        :data="srcData as TypeOf<T>"
+        name="default"
+        :data="srcData as NonNullable<TypeOf<T>>"
         :error="srcError"
         :refresh="props.src !== '' ? refresh : () => {}"
       />
+    </template>
+
+    <template v-else>
       <slot
-        v-if="props.loader && typeof slots.loadable === 'undefined'"
+        v-if="props.loader"
         name="connecting"
         :data="undefined"
         :error="srcError"
         :refresh="props.src !== '' ? refresh : () => {}"
       >
         <XProgress
-          v-if="props.src !== ''"
+          v-if="props.data.length > 0 || props.src !== ''"
```

Comment:
This extra `props.data.length` means XProgress is used correctly even with:

```vue
<DataLoader data="[data]" />
```

type DataLoaders, not just

```vue
<DataLoader src="..." />
```

ones

Previously, only these types were still using the loader from KTable, so all in all this changes nothing from a user perspective but fixes a little non-intentional "bug" in our code.